Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneEventCollector/TrendMicroVisionOneEventCollector.py

```diff
@@ -0,0 +1,1151 @@
+    else:
+        if last_run_next_link:
+            # pagination is over
+            dedup_log_ids = pagination_log_ids
+        last_run_next_link = ''
+
+    search_detections_updated_last_run = {
+        search_detection_start_run_time: last_run_start_time,
+        search_detection_dedup: dedup_log_ids,
+        search_detection_pagination: pagination_log_ids,
+        search_detection_next_link: last_run_next_link
+    }
+
+    demisto.info(f'{search_detection_logs=}')
+    demisto.info(f'{search_detections_updated_last_run=}')
+    return search_detection_logs, search_detections_updated_last_run
+
+
+def get_audit_logs(
+    client: Client,
+    first_fetch: str,
+    last_run: dict,
+    limit: int = DEFAULT_MAX_LIMIT,
+    date_format: str = DATE_FORMAT,
+) -> tuple[List[dict], dict]:
+    """
+    Get the audit logs.
+
+    Args:
+        client (Client): the client object
+        first_fetch (str): the first fetch time
+        last_run (dict): the last run object
+        limit (int): the maximum number of search detection logs to return.
+        date_format (str): the date format.
+
+    Returns:
+        Tuple[List[Dict], Dict]: audit logs & last updated run
+    """
+    audit_cache_time_field_name = LastRunTimeCacheTimeFieldNames.AUDIT.value
+    audit_log_type = LogTypes.AUDIT.value
+    audit_log_last_run_time = LastRunLogsStartTimeFields.AUDIT.value
+
+    start_time, end_time = get_datetime_range(
+        last_run_time=last_run.get(audit_log_last_run_time),
+        first_fetch=first_fetch,
+        log_type_time_field_name=audit_log_last_run_time,
+        date_format=date_format
+    )
+    audit_logs = client.get_audit_logs(
+        start_datetime=start_time, end_datetime=end_time, limit=limit
+    )
+
+    for log in audit_logs:
+        # since there isn't real uuid in audit logs, we hash the entire audit log to create a unique id
+        encoded_audit_log = json.dumps(log, sort_keys=True).encode()
+        log['id'] = hashlib.sha256(encoded_audit_log).hexdigest()
+
+    audit_logs = dedup_fetched_logs(
+        logs=audit_logs,
+        last_run=last_run,
+        log_id_field_name='id',
+        log_cache_last_run_name_field_name=audit_cache_time_field_name,
+        log_type=audit_log_type
+    )
+
+    latest_audit_log_ids, latest_log_time = get_all_latest_logs_ids(
+        logs=audit_logs,
+        log_type=audit_log_type,
+        log_id_field_name='id',
+        date_format=DATE_FORMAT
+    )
+    latest_audit_log_time = latest_log_time or end_time
+    latest_audit_log_time = (dateparser.parse(latest_audit_log_time) - timedelta(seconds=1)).strftime(DATE_FORMAT)  # type: ignore
+
+    for log in audit_logs:
+        # pop all the hashes used to find duplicates
+        log.pop('id', None)
+
+    audit_updated_last_run = {
+        audit_log_last_run_time: latest_audit_log_time,
+        audit_cache_time_field_name: latest_audit_log_ids
+    }
+
+    demisto.info(f'{audit_logs=}')
```

Comment:
same here. first line should be debug

Reply:
https://github.com/demisto/content/pull/26650/files#r1250301598